src/Validation/Validation.php

```diff
@@ -781,6 +784,48 @@ public static function email(mixed $check, ?bool $deep = false, ?string $regex =
         return false;
     }
 
+    /**
+     * Checks that the value is a valid backed enum instance or value.
+     *
+     * @param mixed $check Value to check
+     * @param class-string<\BackedEnum> $enumClassName The valid backed enum class name
+     * @return bool Success
+     * @since 5.1.0
+     */
+    public static function enum(mixed $check, string $enumClassName): bool
+    {
+        if (
+            $check instanceof $enumClassName &&
+            $check instanceof BackedEnum
+        ) {
+            return true;
+        }
+
+        if (
+            !is_string($check) &&
+            !is_int($check)
+        ) {
+            return false;
+        }
+
+        try {
+            $reflectionEnum = new ReflectionEnum($enumClassName);
+            $backingType = (string)$reflectionEnum->getBackingType();
+        } catch (ReflectionException) {
+            return false;
+        }
+
+        if (get_debug_type($check) !== $backingType) {
+            return false;
+        }
+
+        if ($enumClassName::tryFrom($check) !== null) {
+            return true;
+        }
+
+        return false;
```

Comment:
```suggestion
        return $enumClassName::tryFrom($check) !== null;
```

We can golf this down a bit :golfing:

Reply:
My excuse is as always code coverage visibility 🙃

Reply:
Explicitly checking the backing type does seem redundant to me when `tryFrom()` would fail anyway if a value of incorrect type is passed. We can add a comment to related assertion stating it's checking for incorrect type :)

Reply:
@ADmad You mean the `get_debug_type()` check? I guess it's somewhat redundant, I left that in because `tryFrom()` is relatively slow. But the `get_debug_type()` check is only about twice as fast, and the real killer is reflection, so I guess that bailout isn't too helpful.

Reply:
@ADmad Actually, that check also prevents a possible type error, so the `tryFrom()` would need to be wrapped in a `try`. That's IMHO kinda stupid of PHP, I would've expected that `tryFrom()` prevents exactly that kind of exception, even with `strict_types` enabled 🤔

Reply:
And turns out it's really, really, really slow, just that type error alone costs more than double the time the whole method with reflection and stuff takes.